packages/twenty-server/src/engine/core-modules/graphql/utils/graphql-errors.util.ts

```diff
@@ -113,94 +115,266 @@ export class ValidationError extends BaseGraphQLError {
   }
 }
 
+export class NotFoundError extends BaseGraphQLError {
+  constructor(customException: CustomException);
+
+  constructor(message: string, extensions?: RestrictedGraphQLErrorExtensions);
+
+  constructor(
+    messageOrException: string | CustomException,
+    codeOrExtensions?: string | RestrictedGraphQLErrorExtensions,
+  ) {
+    if (messageOrException instanceof CustomException) {
+      const exception = messageOrException;
+
+      super(exception.message, ErrorCode.NOT_FOUND, {
+        userFriendlyMessage: exception.userFriendlyMessage,
+        subCode: exception.code,
+      });
+    } else {
+      const message = messageOrException as string;
+      const extensions = codeOrExtensions as RestrictedGraphQLErrorExtensions;
+
+      super(message, ErrorCode.NOT_FOUND, extensions);
+    }
+    Object.defineProperty(this, 'name', { value: 'NotFoundError' });
+  }
+}
+
 export class AuthenticationError extends BaseGraphQLError {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  constructor(message: string, extensions?: RestrictedGraphQLErrorExtensions) {
-    super(message, ErrorCode.UNAUTHENTICATED, extensions);
+  constructor(customException: CustomException);
```

Comment:
I'm never a fan of overloading APIs when there is no real need. Is this for migration purprose? If yes, let's add a TODO